File: backend/src/account/lrembecki.core.account/ViewModels/AccountSubscriptionVM.cs
using lrembecki.core.account.Entities;

namespace lrembecki.core.account.ViewModels;

public record AccountSubscriptionVM(
    Guid Id,
    string Email,
    string Subscription,
    Guid SubscriptionId,
    bool IsActive,
    bool IsDefault,
    ICollection<PermissionGroupVM> PermissionGroups
)
{
    public static AccountSubscriptionVM Map(
        AccountSubscriptionEntity accountSubscriptionEntity
    )
    {
        if (accountSubscriptionEntity == null) return null!;

        return new AccountSubscriptionVM(
            accountSubscriptionEntity.Id,
            accountSubscriptionEntity.Account.Email,
            accountSubscriptionEntity.Subscription.Name,
            accountSubscriptionEntity.SubscriptionId,
            accountSubscriptionEntity.IsActive,
            accountSubscriptionEntity.IsDefault,
            accountSubscriptionEntity.PermissionGroups
                .Select(PermissionGroupVM.Map)
                .ToList()
        );
    }
}

File: frontend/src/app/modules/administration/subscription-accounts/account-subscription.vm.ts
import { PermissionGroupVM } from '../permission-groups/permission-group.vm';

export class AccountSubscriptionVM {
  id: string = null!;
  email: string = null!; // account email
  subscription: string = null!; // subscription name/code
  isActive: boolean = false;
  isDefault: boolean = false;
  permissionGroups: PermissionGroupVM[] = [];

  constructor(init?: Partial<AccountSubscriptionVM>) {
    Object.assign(this, init);
    this.permissionGroups = init?.permissionGroups?.map(p => new PermissionGroupVM(p)) || [];
  }
}
